File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    while len(goal) :
        if len(cards1) > 0 and cards1[0] == goal[0] :
            cards1.pop(0)
        elif len(cards2) > 0 and cards2[0] == goal[0]:
            cards2.pop(0)
        else :
            break
        goal.pop(0)
        
    return len(goal) > 0 and "No" or "Yes"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cs
using System;
using System.Collections.Generic;

public class Solution {
    public string solution(string[] cards1, string[] cards2, string[] goal) {
        Queue<string> cardQueue1 = new Queue<string>(cards1);
        Queue<string> cardQueue2 = new Queue<string>(cards2);
        string answer = "Yes";
        for(int i=0; i<goal.Length; ++i)
        {
            if(cardQueue1.Count > 0 && cardQueue1.Peek() == goal[i])
            {
               cardQueue1.Dequeue(); 
            }
            else if(cardQueue2.Count > 0 && cardQueue2.Peek() == goal[i])
            {
                cardQueue2.Dequeue();
            }
            else
            {
                answer = "No";
                break;
            }
        }
        
        return answer;
    }
}